Firestore/Source/Public/FIRFirestoreSettings.h

```diff
@@ -44,6 +44,18 @@ NS_SWIFT_NAME(FirestoreSettings)
 /** Set to false to disable local persistent storage. */
 @property(nonatomic, getter=isPersistenceEnabled) BOOL persistenceEnabled;
 
+/**
+ * If true, FIRDocumentSnapshots will return timestamp fields as FIRTimestamps. If false, legacy
+ * behavior will be used: timestamp fields will be returned as NSDates, which may lead to loss of
+ * precision due to conversion to double.
+ *
+ * Defaults to false. NOTE: at some future point, enableTimestampsInSnapshots will switch to true by
+ * default, and eventually this property will be removed altogether. Currently, a warning will be
+ * logged if enableTimestampsInSnapshots is set to false. Please upgrade your code to expect
+ * FIRTimestamp instead of NSDate and set enableTimestampsInSnapshots to true.
+ */
+@property(nonatomic) BOOL enableTimestampsInSnapshots;
```

Comment:
Any suggestions about the wording are welcome.

Reply:
Here's my suggestion:

> Enables the use of FIRTimestamps for timestamp fields in FIRDocumentSnapshots.
> 
> Currently, Firestore returns timestamp fields as an `NSDate` but `NSDate` is implemented as a `double` which loses precision and causes unexpected behavior when using a timestamp from a snapshot as a part of a subsequent query.
>
> Setting `enableTimestampsInSnapshots` to `true` will cause Firestore to return `FIRTimestamp` values instead of `NSDate`, avoiding this kind of problem. To make this work you must also change any code that uses `NSDate` to use `FIRTimestamp` instead.
>
> NOTE: in the future `enableTimestampsInSnapshots = true` will become the default and this option will be removed so you should change your code to use FIRTimestamp now and opt-in to this new behavior as soon as you can.

Reply:
nit: `true` should be `YES` if we're using Obj-C syntax (which we seem to be with `FIRTimestamp` etc.)

Reply:
Gil, changed to use your wording.

Michael, I was originally following the comment on `persistenceEnabled`, which used `false`. I don't mind either way, but I think this should be consistent. Let's change both?

Reply:
Ahh, I see. My preference would be to ultimately move towards Swift syntax since that's the dominant language of our users (which would mean changing `FIRTimestamp` to `Timestamp` or maybe "Firebase `Timestamp`" in this case since "Timestamp" could be ambiguous on its own).

But I don't think we should try to tackle that in this PR. :-) So I think leaving this as-is is fine.